src/adapter/src/coord.rs

```diff
@@ -3138,6 +3138,34 @@ impl<S: Append + 'static> Coordinator<S> {
                 },
             )
             .await;
+
+        let storage_sink_from_entry = self.catalog.get_entry(&sink.from);
```

Comment:
(can't find your comment to reply inline idk?)

you're right that it is duplicative! this is the part where I wasn't sure the best way to make sure the "storaged way" continues to compile without totally breaking sinks (which would happen if I removed the above code) -- so I settled on a little duplication.

Would you be satisfied if I moved this part out to a function that we don't yet call / hit it with a `#[allow(dead_code)]`?

edit: but oh yes you're very correct whoops -- `handle_sink_connection_ready` is the right place to put this I think